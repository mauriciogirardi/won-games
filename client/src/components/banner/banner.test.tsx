import { screen } from '@testing-library/react'

import { renderWithTheme } from '@/utils/tests/helpers'
import { Banner } from '.'

const bannerProps = {
  img: 'https://source.unsplash.com/user/mauriciogirardi/1042x580',
  title: 'Defy death',
  subtitle: '<p>Play the new <strong>CrashLands</strong> season</p>',
  buttonLink: '/games/defy-death',
  buttonLabel: 'Buy now'
}

describe('<Banner />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<Banner {...bannerProps} />)

    expect(
      screen.getByRole('heading', { name: /defy death/i })
    ).toBeInTheDocument()
    expect(screen.getByRole('link', { name: /buy now/i })).toHaveAttribute(
      'href',
      '/games/defy-death'
    )
    expect(
      screen.getByRole('heading', { name: /Play the new CrashLands season/i })
    ).toBeInTheDocument()

    expect(screen.getByRole('img', { name: /defy death/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render a Ribbon', () => {
    renderWithTheme(
      <Banner
        {...bannerProps}
        ribbon={{
          name: 'May Ribbon',
          size: 'small',
          color: 'secondary'
        }}
      />
    )

    const ribbon = screen.getByText(/may ribbon/i)

    expect(ribbon).toBeInTheDocument()
    expect(ribbon).toHaveStyle({ backgroundColor: '#3CD3C1' })
    expect(ribbon).toHaveStyle({
      height: '2.6rem',
      fontSize: '1.2rem'
    })
  })
})
