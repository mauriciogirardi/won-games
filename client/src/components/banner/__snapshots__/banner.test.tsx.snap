// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> should render correctly 1`] = `
.c5 {
  background: linear-gradient(180deg, #ff5f5f 0%, #f062c0 50%);
  border-radius: 0.4rem;
  border: 0;
  color: #FAFAFA;
  padding: 0.8rem;
  cursor: pointer;
  overflow: hidden;
  text-decoration: none;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  outline: none;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
  height: 4rem;
}

.c5:focus {
  box-shadow: 0 0 0.5rem #F231A5;
}

.c5:hover {
  background: linear-gradient(180deg, #e35565 0%, #d958a6 50%);
}

.c0 {
  position: relative;
}

.c1 {
  width: 100%;
  height: 23rem;
  background-color: #EAEAEA;
  background-image: url(https://source.unsplash.com/user/mauriciogirardi/1042x580);
  background-position: center center;
  background-size: cover;
}

.c2 {
  width: 100%;
  background-color: rgba(0, 0, 0, 0.7);
  padding: 2.4rem;
}

.c3 {
  font-size: 1.8rem;
  font-weight: 600;
  color: #FAFAFA;
}

.c4 {
  font-size: 1.4rem;
  font-weight: 400;
  color: #FAFAFA;
  margin-bottom: 1.6rem;
}

.c4 strong {
  color: #F231A5;
}

@media (max-width: 1170px) {

}

@media (min-width: 768px) {
  .c0 {
    box-shadow: 0 0.4rem 0.5rem 0 rgba(0,0,0,0.2);
  }
}

@media (min-width: 768px) {
  .c1 {
    height: 58rem;
    border-radius: 0.4rem;
  }
}

@media (min-width: 768px) {
  .c2 {
    border-radius: 0 0 0.4rem 0.4rem;
    padding: 4.0rem;
    position: absolute;
    bottom: 0;
    left: 0;
  }
}

@media (min-width: 768px) {
  .c3 {
    font-size: 2.8rem;
  }
}

@media (min-width: 768px) {
  .c4 {
    font-size: 1.8rem;
  }
}

<main
  class="c0"
>
  <div
    aria-label="Defy death"
    class="c1"
    role="img"
  />
  <div
    class="c2"
  >
    <h2
      class="c3"
    >
      Defy death
    </h2>
    <h3
      class="c4"
    >
      <p>
        Play the new 
        <strong>
          CrashLands
        </strong>
         season
      </p>
    </h3>
    <a
      class="c5"
      href="/games/defy-death"
    >
      <span>
        Buy now
      </span>
    </a>
  </div>
</main>
`;
